travis_cargo.py

```diff
@@ -173,6 +174,10 @@ def raw_coverage(use_sudo, verify, test_args, merge_msg, kcov_merge_args, kcov_m
     # inconsistent/inaccurate, so using the output of read-manifest is
     # far too much trouble.
 
+    if link_dead_code:
+        flags = ','.join((os.environ.get('RUSTFLAGS', ''), '-C link-dead-code'))
+        os.environ['RUSTFLAGS'] = flags
+
     output = run_output('cargo', 'test', *test_args)
```

Comment:
This should be adjusted to have an `env = {'RUSTFLAGS': flags}` argument... although I realise the modification I mention only applied to `run`, so `run_output` should probably be adjusted to be similar to the new `run`.

I think we'll want to run `cargo clean` first too: IME, cargo doesn't recompile things when `RUSTFLAGS` changes (maybe a bug in cargo, but travis-cargo has to deal with it).